### ISSUE TITLE
Add a FindPeerAsync function

### ISSUE THREAD

**Issue by Stebalien** [libp2p/go-libp2p-routing#13]
There's a FindProvidersAsync, there will soon be a GetValuesAsync, but there is no GetPeerAsync. With a GetPeerAsync, we could try peer addresses as we find them instead of waiting for the search to complete. Benefits over the current system:

* Can abort searches early (save bandwidth)
* Can keep searching until we find an address for the peer that works (reliability).

**Comment by Kubuxu** [comment]
Yes, it would be quite useful as (IIRC) it would remove one layer of workers from Bitswap.

**Comment by Stebalien** [comment]
My usecase is implementing a parallel multirouter (i.e., a router that queries multiple sub-routers in parallel).
